src/components/Widget/index.tsx

```diff
@@ -32,19 +39,29 @@ export default function Widget({ defaultToken, onReviewSwapClick }: WidgetProps)
   const { settings } = useSyncWidgetSettings()
   const { transactions } = useSyncWidgetTransactions()
 
+  const onSwitchChain = useCallback(
+    // TODO: Widget should not break if this rejects - upstream the catch to ignore it.
+    ({ chainId }: AddEthereumChainParameter) => switchChain(connector, Number(chainId)).catch(() => undefined),
+    [connector]
+  )
+
+  if (!inputs.value.INPUT && !inputs.value.OUTPUT) {
```

Comment:
where would this happen? i think this makes sense on token details but not sure it makes sense elsewhere (not that we would use it elsewhere?)

Reply:
Right now the widget is specialized to token details, and is not used elsewhere.
I'd expect to break it into its own component before using it elsewhere.